drivers/platform/surface/surface_aggregator_registry.c

```diff
@@ -68,12 +68,26 @@ static const struct software_node ssam_node_bat_sb3base = {
 	.parent = &ssam_node_hub_base,
 };
 
-/* Platform profile / performance-mode device. */
-static const struct software_node ssam_node_tmp_pprof = {
+/* Platform profile / performance-mode device without a fan. */
+static const struct software_node ssam_node_perf_profile = {
```

Comment:
So I thought about this a bit and I'd prefer to keep the `tmp` prefix as indicator for the subsystem in the registry. So `ssam_node_SUBSYS_NAME`, e.g., `ssam_node_tmp_perf_profile`. I'll add a comment for this to the fan monitor patches you submitted to the mailing list as well.

Reply:
yeah, that's a good idea actually will change that over.

Reply:
Just to mention it here as well: I somehow misremembered and though the fan was in the TMP category, not FAN. So disregard the part mentioning the fan monitor patches. That's all fine.

Reply:
So what naming convention do you want to follow if it has two subsystems? :D 

If we use the subsystem naming convention, I guess we end up with: 
- `ssam_node_tmp_perf_profile` (instead the original `ssam_node_tmp_pprof`)
- `ssam_node_tmp_fan_perf_profile` (the new thing, currently named `ssam_node_perf_profile_with_fan`)

Reply:
Let's stick with `tmp` for this one (i.e. `ssam_node_tmp_perf_profile`), since it's still kind of the main thing.

Reply:
So for the non-fan version `ssam_node_tmp_perf_profile` and for the fan version `ssam_node_tmp_perf_profile_with_fan`?

I'm actually not sure what the TMP profile actually does... we should check some clock speeds or something to determine if it does anything at all >_<

Reply:
Yeah, looks good.

On the SB2 it changes the fan profile of the dGPU fan. But that's about all that I could figure out. I doubt it changes any clocks. SAM shouldn't have access that direct to the CPU, I think.

Reply:
> SAM shouldn't have access that direct to the CPU, I think.

Hmm, I'm not so sure of that, I found the string `IMVP MP2964 (U6601)` in the firmware blob for it... which definitely looks like [this chip](https://www.monolithicpower.com/en/documentview/productdocument/index/version/2/document_type/Datasheet/lang/en/sku/MP2965/document_id/5134/) that is specificially made to control power to the CPU, and I wouldn't be surprised if one of the 'low power' modes drops the voltage of the cpu down a bit to force it to reduce the clock speed? Like that would work on a microcontroller at least, I'm not sure if that  can be done on a modern CPU...

There's also `Batt Charger BQ25713 (U6302)` in the firmware, which seems like an interesting [IC](https://www.ti.com/lit/ds/symlink/bq25713.pdf) as it provides access to various currents, which would allow some insight into whether low power mode is actually lower power... I tried to find that, but just loading `i2c-dev` resulting in a bunch of i2c busses, but none of them (except the display one if I remember correctly) had anything on it that answered back. I part expect that we don't actually have access to the useful i2c bus... because that's what the management ic is for :(

Reply:
Incorporated.

Reply:
> > SAM shouldn't have access that direct to the CPU, I think.
> 
> Hmm, I'm not so sure of that, I found the string `IMVP MP2964 (U6601)` in the firmware blob for it... which definitely looks like [this chip](https://www.monolithicpower.com/en/documentview/productdocument/index/version/2/document_type/Datasheet/lang/en/sku/MP2965/document_id/5134/) that is specificially made to control power to the CPU, and I wouldn't be surprised if one of the 'low power' modes drops the voltage of the cpu down a bit to force it to reduce the clock speed? Like that would work on a microcontroller at least, I'm not sure if that can be done on a modern CPU...

Hmm interesting... The reason I doubted that SAM would have access is that there's already a software-interface for controlling the clock speeds/power limits on modern CPUs. I would have assumed that Windows uses that rather than go through SAM somehow.

Reply:
Yeah, that's probably the 'main' thing, but then the question is what does the whole TMP module still do when we switch profiles :thinking: